src/Rules/ValidEvenNumber.php

```diff
@@ -11,7 +11,20 @@ class ValidEvenNumber implements Rule
      */
     public function passes($attribute, $value): bool
     {
-        return preg_match('/^\d*[02468]$/', $value);
+        $number = strval($value);
+        $number = explode('.', $number);
+
+        if (isset($number[1]) && $number[1] != 0) {
+            return false;
+        }
+
+        $number = $number[0];
+
+        if (extension_loaded('gmp')) {
```

Comment:
Well, I arrived late here 😄.
I suggest adding the **gmp** extension to the suggest section of the **composer.json** file and to the GitHub Actions settings for installation in the CI/CD environment:

```json
"suggest": {
    "ext-gmp": "Required for handling large numbers efficiently."
}
```

https://github.com/milwad-dev/laravel-validate/blob/bc3faaccc29e82a178dd00b45e4185d032351e8d/.github/workflows/run-tests.yml#L50

```diff
+extensions: curl, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, iconv, gmp
```